spec-tests/src/test/java/org/candlepin/spec/pools/PoolUnlimitedMasterSpecTest.java

```diff
@@ -0,0 +1,341 @@
+        guestUnmapped = Consumers.random(owner)
+            .name(StringUtil.random("host"))
+            .type(ConsumerTypes.System.value())
+            .facts(Map.of("virt.is_guest", "true"));
+        guestUnmapped = consumerApi.createConsumer(guestUnmapped);
+        guestUnmappedClient = ApiClients.trustedConsumer(guestUnmapped.getUuid());
+
+        productNoVirt = Products.randomSKU()
+            .addAttributesItem(new AttributeDTO().name("multi-entitlement").value("yes"));
+        productUnlimitedVirt = Products.randomSKU()
+            .addAttributesItem(new AttributeDTO().name("virt_limit").value("unlimited"))
+            .addAttributesItem(new AttributeDTO().name("multi-entitlement").value("yes"));
+        productVirt = Products.randomSKU()
+            .addAttributesItem(new AttributeDTO().name("virt_limit").value("5"))
+            .addAttributesItem(new AttributeDTO().name("multi-entitlement").value("yes"));
+        proudctVirtHostDep = Products.randomSKU()
+            .addAttributesItem(new AttributeDTO().name("virt_limit").value("5"))
+            .addAttributesItem(new AttributeDTO().name("multi-entitlement").value("yes"))
+            .addAttributesItem(new AttributeDTO().name("host-dependent").value("true"));
+        productVirtInstanceMuliplier = Products.randomSKU()
+            .addAttributesItem(new AttributeDTO().name("virt_limit").value("8"))
+            .addAttributesItem(new AttributeDTO().name("multi-entitlement").value("yes"))
+            .addAttributesItem(new AttributeDTO().name("instance_multiplier").value("6"));
+        productVirtProductMuliplier = Products.randomSKU()
+            .addAttributesItem(new AttributeDTO().name("virt_limit").value("8"))
+            .addAttributesItem(new AttributeDTO().name("multi-entitlement").value("yes"))
+            .multiplier(100L);
+
+        productNoVirt = ownerProductApi.createProductByOwner(owner.getKey(), productNoVirt);
+        productUnlimitedVirt = ownerProductApi.createProductByOwner(owner.getKey(), productUnlimitedVirt);
+        productVirt = ownerProductApi.createProductByOwner(owner.getKey(), productVirt);
+        proudctVirtHostDep = ownerProductApi.createProductByOwner(owner.getKey(), proudctVirtHostDep);
+        productVirtInstanceMuliplier = ownerProductApi.createProductByOwner(owner.getKey(),
+            productVirtInstanceMuliplier);
+        productVirtProductMuliplier = ownerProductApi.createProductByOwner(owner.getKey(),
+            productVirtProductMuliplier);
+
+        poolNoVirt = createUnlimitedPool(productNoVirt);
+        poolUnlimitedVirt = createUnlimitedPool(productUnlimitedVirt);
+        poolVirt = createUnlimitedPool(productVirt);
+        poolVirtHostDep = createUnlimitedPool(proudctVirtHostDep);
+        poolVirtInstanceMuliplier = createUnlimitedPool(productVirtInstanceMuliplier);
+        poolVirtProductMuliplier = createUnlimitedPool(productVirtProductMuliplier);
```

Comment:
It looks like most of the setup is not actually needed for every test. Most of the tests require only 1-2 pools and don't care about the rest. It could be beneficial to split it up.

The only place that actually needs all of them is `shouldHaveCreatedCorrectPools`, but I think those assertions could be split and integrated into other tests as well.